crm_lead_type/models/crm_lead.py

```diff
@@ -0,0 +1,10 @@
+# Copyright (C) 2020 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class CrmLead(models.Model):
+    _inherit = "crm.lead"
+
+    lead_type_id = fields.Many2one("crm.lead.type", string="Lead Type")
```

Comment:
Maybe adding ondelete="restrict"?

Do you want to implement default lead_type logic like in sale_order_type addon (store default lead_type at res.partner)?

Reply:
Restrict sounds good.
Adding default logic is an interesting idea, but would require wither a glue module or `some changes to `sale_order_type.
I would leave that our for know until I'm sure the customer using this needs that.

Reply:
Okay, no problem, thank you for your feedback.
Also noticed that the crm_lead_type field is only visible for type = lead, not for type = opportunity (https://github.com/odoo/odoo/blob/15269bcfa8e4d914b2834334b6ff797c5538a400/addons/crm/views/crm_lead_views.xml#L222)

Is this the correct behavior?